tools/network.py

```diff
@@ -68,6 +95,50 @@ def fetch_raw(
     :param params: Dictionary to send in the query string
     :return: bytes of the content and default name of the file or empty string
     """
+    return request_raw(url, "get", encoding, authcfg_id, params)
+
+
+def post_raw(
+    url: str,
+    encoding: str = ENCODING,
+    authcfg_id: str = "",
+    data: Optional[Dict[str, str]] = None,
+    files: Optional[Dict[str, bytes]] = None,
+) -> Tuple[bytes, str]:
+    """
+    Post resource. Similar to requests.post(url, data) but is
+    recommended way of handling requests in QGIS plugin
+    :param url: address of the web resource
+    :param encoding: Encoding which will be used to decode the bytes
+    :param authcfg_id: authcfg id from QGIS settings, defaults to ''
+    :param data: Dictionary to send in the request body
+    :param files: Files to send multipart-encoded
+    :return: bytes of the content and default name of the file or empty string
+    """
+    return request_raw(url, "post", encoding, authcfg_id, None, data, files)
+
+
+def request_raw(
+    url: str,
+    method: str = "get",
```

Comment:
This could be changed to Enum. Or at least Type should be `Literal["get", "post"]` to make clear with supported values.